File: src/custom_msgs/src/py/cm_screen.py
#!/usr/bin/env python
import rospy
from rospy.exceptions import ROSInterruptException

# Import custom messages
from custom_msgs.msg import IoTSensor

# Display the data to stdout
def display_data(data_point):
	print(f'''
id: {data_point.id}
name: {data_point.name}
temperature: {data_point.temperature}
humidity: {data_point.humidity}
	''')

# Define node action
def node():
	# Initialize the node
	rospy.init_node("cm_screen", anonymous=False)

	# Create a subscriber
	sub = rospy.Subscriber("cm_data", IoTSensor, display_data)

	# Keep listening
	rospy.spin()

# Start the node when the file is executed
if __name__ == "__main__":
	try:
		node()
	except ROSInterruptException:
		pass

File: src/custom_msgs/src/cpp/cm_screen.cpp
#include <ros/ros.h>

// Include custom message
#include "custom_msgs/IoTSensor.h"

void callback(const custom_msgs::IoTSensor& data_point) {
  printf("id: %d\nname: %s\ntemperature: %lf\nhumidity: %lf\n",
         data_point.id,
         data_point.name.c_str(),
         data_point.temperature,
         data_point.humidity);
}

// Define node action
int main(int argc, char* argv[]) {
  // Initialize the node
  ros::init(argc, argv, "cm_screen");

  // Create a subscriber
  ros::NodeHandle node;
  ros::Subscriber sub = node.subscribe("cm_data", 10, callback);

  // Keep listening
  ros::spin();

  return 0;
}
